pallets/xcmp-handler/src/lib.rs

```diff
@@ -355,6 +375,59 @@ pub mod pallet {
 			Ok((local_xcm, target_xcm))
 		}
 
+		/// Construct the alternate xcm flow instructions
+		///
+		/// There are no local instructions since the user's account is already funded on the target chain
+		///
+		/// Target instructions
+		/// 	- DescendOrigin
+		///     - WithdrawAsset
+		/// 	- BuyExecution
+		/// 	- Transact
+		/// 	- RefundSurplus
+		/// 	- DepositAsset
+		fn get_alternate_flow_instructions(
+			para_id: ParachainId,
+			currency_id: T::CurrencyId,
+			descend_location: Junctions,
+			transact_encoded_call: Vec<u8>,
+			transact_encoded_call_weight: u64,
+			xcm_weight: u64,
+			fee: u128,
+		) -> Result<
+			(xcm::latest::Xcm<<T as pallet::Config>::Call>, xcm::latest::Xcm<()>),
+			DispatchError,
+		> {
+			let target_asset_loc = T::CurrencyIdToMultiLocation::convert(currency_id)
+				.ok_or(Error::<T>::CurrencyNotSupported)?;
+			let target_asset =
```

Comment:
I'm not confident about this asset location specified in this way. We'll need to test it once more of the pieces are in place.